types/types.go

```diff
@@ -689,7 +781,7 @@ func (s *PrimitiveType[T]) MatchWithoutNullability(ot Type) bool {
 	return false
 }
 
-func (s *PrimitiveType[T]) ReturnType() (Type, error) {
+func (s *PrimitiveType[T]) ReturnType([]FuncDefArgType, []Type) (Type, error) {
```

Comment:
not sure what this signature change is throughout the patch.

Reply:
Raised PR #80 for return type derivation